cmd/embedded-cluster/install.go

```diff
@@ -179,7 +181,73 @@ func checkLicenseMatches(c *cli.Context) error {
 	}
 
 	return nil
+}
+
+func checkAirgapMatches(c *cli.Context) error {
+	rel, err := release.GetChannelRelease()
+	if err != nil {
+		return fmt.Errorf("failed to get release from binary: %w", err) // this should only be if the release is malformed
+	}
+	if rel == nil {
+		return nil // TODO make this an error when done testing
+	}
+
+	// read file from path
+	rawfile, err := os.Open(c.String("airgap-bundle"))
+	if err != nil {
+		return fmt.Errorf("failed to open airgap file: %w", err)
+	}
+	defer rawfile.Close()
+
+	appSlug, channelID, airgapVersion, err := airgap.AirgapBundleVersions(rawfile)
+	if err != nil {
+		return fmt.Errorf("failed to get airgap bundle versions: %w", err)
+	}
+
+	// Check if the airgap bundle matches the application version data
+	if rel.AppSlug != appSlug {
+		// if the app is different, we will not be able to provide the correct vendor supplied charts and k0s overrides
+		return fmt.Errorf("airgap bundle app %s does not match binary app %s, please provide the correct bundle", appSlug, rel.AppSlug)
+	}
+	if rel.ChannelID != channelID {
+		// if the channel is different, we will not be able to install the pinned vendor application version within kots
+		return fmt.Errorf("airgap bundle channel %s does not match binary channel %s, please provide the correct bundle", channelID, rel.ChannelID)
+	}
+	if rel.VersionLabel != airgapVersion {
```

Comment:
Hmm, should this check channel sequence instead of version label? As version labels aren't always unique. Might be an edge case though.

Reply:
We're not compiling in the channel sequence right now is the problem

Reply:
(we could, but it would still leave the problem of "semver mismatch makes for a far better error message than inscrutable sequence numbers")